server/server.go

```diff
@@ -530,15 +530,8 @@ func (s *Server) Kill(connectionID uint64, query bool) {
 
 func killConn(conn *clientConn) {
 	conn.mu.RLock()
-	resultSets := conn.mu.resultSets
 	cancelFunc := conn.mu.cancelFunc
 	conn.mu.RUnlock()
-	for _, resultSet := range resultSets {
-		// resultSet.Close() is reentrant so it's safe to kill a same connID multiple times
-		if err := resultSet.Close(); err != nil {
```

Comment:
As we won't call `Close` now, then, how to make sure the executor stop running and release resource?

Reply:
Check `context.Contex` in every `executor.Next`?

Reply:
It works like this:
tidb receive `kill` input =>
session manager call the cancel function =>
the goroutine of the killed session run to the `Next()` function =>
check ctx.Done() in `Next()` and return error =>
the caller find `Next()` return error, it call `Close()` =>
finish current transaction